receiver/prometheusreceiver/config.go

```diff
@@ -93,6 +163,42 @@ func (cfg *Config) Validate() error {
 				return fmt.Errorf("error validating scrapeconfig for job %v: %w", sc.JobName, errRenamingDisallowed)
 			}
 		}
+
+		if sc.HTTPClientConfig.Authorization != nil {
+			if err := checkFile(sc.HTTPClientConfig.Authorization.CredentialsFile); err != nil {
+				return fmt.Errorf("error checking authorization credentials file %q - %s", sc.HTTPClientConfig.Authorization.CredentialsFile, err)
+			}
+		}
+
+		if err := checkTLSConfig(sc.HTTPClientConfig.TLSConfig); err != nil {
+			return err
+		}
+
+		for _, c := range sc.ServiceDiscoveryConfigs {
+			switch c := c.(type) {
+			case *kubernetes.SDConfig:
+				if err := checkTLSConfig(c.HTTPClientConfig.TLSConfig); err != nil {
+					return err
+				}
+			case *file.SDConfig:
+				for _, file := range c.Files {
+					files, err := filepath.Glob(file)
+					if err != nil {
+						return err
+					}
+					if len(files) != 0 {
+						for _, f := range files {
+							err = checkSDFile(f)
+							if err != nil {
+								return fmt.Errorf("checking SD file %q: %v", file, err)
+							}
+						}
+						continue
+					}
+					fmt.Printf("WARNING: file %q for file_sd in scrape job %q does not exist\n", file, sc.JobName)
```

Comment:
Should we be returning an error here instead of printing a warning?  If we still want to print, should we be using a logger?

Reply:
Makes sense. I have updated it to return error instead of logging to keep it consistent. Have fixed another test which had a non existent file here -https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/5052/files#diff-91a13efe7ac836919f7e9db1752692243725fbf17683c645dedcb27a6cfff4ac  Please take a look.